rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -1152,66 +1271,379 @@ const auto ppu_stwcx_tx = build_function_asm<u32(*)(u32 raddr, u64 rtime, u64 rd
 	c.ret();
 });
 
-const auto ppu_stdcx_tx = build_function_asm<u32(*)(u32 raddr, u64 rtime, u64 rdata, u64 value)>([](asmjit::X86Assembler& c, auto& args)
+const auto ppu_stdcx_accurate_tx = build_function_asm<u32(*)(u32 raddr, u64 rtime, const void* _old, u64 _new)>([](asmjit::X86Assembler& c, auto& args)
 {
 	using namespace asmjit;
 
 	Label fall = c.newLabel();
 	Label fail = c.newLabel();
+	Label _ret = c.newLabel();
+	Label skip = c.newLabel();
+	Label next = c.newLabel();
+
+	if (utils::has_avx() && !s_tsx_avx)
```

Comment:
utils::has_avx() and s_tsx_avx store the same value - processor capability. It's never changed in runtime. So this check is never true due to boolean rules (A && !A = false).
Seen this in PPUThread, SPUThread, PPUTranslator and similar files.